source/components/ch4_component.cpp

```diff
@@ -104,25 +104,22 @@ void CH4Component::setData( const string& varName,
     try {
          if( varName ==  D_PREINDUSTRIAL_CH4  ) {
             H_ASSERT( data.date == Core::undefinedIndex() , "date not allowed" );
-            M0 = unitval::parse_unitval( data.value_str, data.units_str, U_PPBV_CH4 );
```

Comment:
Ok, here are my thoughts on this:

1. We _definitely_ want to allow a `unitval` initialized with a double and a unit to be passed through `sendMessage`.  If we can't do that already, then that's a bug (and thanks for finding and fixing it!)
2. We _don't_ want to allow a caller to pass a `unitval` initialized without the correct unit to through `sendMessage`.  Our philosophy on units is that you don't have to use Hector's unit checking system to interface with Hector, but you do have to show that whoever programmed the interface knew what the expected unit was (and, equally importantly, that the unit hasn't changed in some recent update -- I won't mention the name of the teammate that changed the units from under me a few years back).  <p> Now, it's true that we make an exception to this philosophy for values passed in as strings.  Arguably we shouldn't do that, but values passed as strings _should_ come exclusively from input parsing, and parsing a value and a unit complicates the parsing a lot, especially when the value is coming from a csv table (because in this case the unit appears only in the header line of the table, so you have to match it up with each row).  Arguably we should bite the bullet and parse units on input anyhow because right now if we were to change the unit on an input quantity in the code, the system would silently parse input with the wrong units.  However, fixes like that have taken a back seat to actual science improvements.  You know how it is.
3. You mentioned passing `unitval`s to `setData`.  The `setData` method really shouldn't be called from outside the component.  The fact that it is public instead of private is a holdover from Hector's early development, before the core and its message routing were established.  There has been a TODO comment in that method suggesting that we track down and eliminate the remaining external usages, but it also hasn't been a priority for the same reasons as discussed above.  In practical terms, however, any new code should communicate with Hector components via `Core::sendMessage`.  If there is some use case that we can't support with `sendMessage`, then we should flag that so we can fix the missing functionality, rather than circumventing the intended communication pathway.

Does that clear things up at all?